wallet/src/client.rs

```diff
@@ -67,33 +69,32 @@ pub fn send_partial_tx(url: &str, partial_tx: &PartialTx) -> Result<PartialTx, E
 }
 
 fn single_send_partial_tx(url: &str, partial_tx: &PartialTx) -> Result<PartialTx, Error> {
-	let mut core = reactor::Core::new()?;
+    let mut core = reactor::Core::new().context(ErrorKind::Hyper)?;
```

Comment:
We have to be more explicit here? By specifying the `ErrorKind::Hyper` context rather than just using `?`.
I've only taken a quick look through this but I'm not sure I fully understand the benefits that Error/ErrorKind bring here?